sihl/src/contract_user.ml

```diff
@@ -101,73 +126,79 @@ module type Sig = sig
     -> unit
     -> (t, string) Result.t Lwt.t
 
-  val create
-    :  email:string
-    -> password:string
-    -> username:string option
-    -> admin:bool
-    -> confirmed:bool
-    -> (t, string) result Lwt.t
-
-  (** Create and store a user. *)
+  (** [create_user ?username ?name ?given_name email password] returns a
+      non-admin user. Note that using [create_user] skips the registration
+      workflow and should only be used with care.*)
   val create_user
-    :  email:string
-    -> password:string
-    -> username:string option
+    :  ?username:string
+    -> ?name:string
+    -> ?given_name:string
+    -> string
+    -> string
     -> t Lwt.t
 
-  (** Create and store a user that is also an admin. *)
+  (** [create_admin ?username ?name ?given_name email password] returns an admin
+      user. *)
   val create_admin
-    :  email:string
-    -> password:string
-    -> username:string option
+    :  ?username:string
+    -> ?name:string
+    -> ?given_name:string
+    -> string
```

Comment:
Why are these not named anymore? They are both `string` and are next to each other

Reply:
Hmm true, maybe we can have email as `string` next to `password:string`? If passwords get stored as plaintext that is quite bad.

Reply:
Yea that sounds good

Reply:
Fixed.

Reply:
Thanks